src/services/activeLearningService.ts

```diff
@@ -0,0 +1,105 @@
+    private objectDetection: ObjectDetection;
+    private modelLoaded: boolean = false;
+
+    constructor(private settings: IActiveLearningSettings) {
+        Guard.null(settings);
+        this.objectDetection = new ObjectDetection();
+    }
+
+    public isModelLoaded() {
+        return this.modelLoaded;
+    }
+
+    public async predictRegions(canvas: HTMLCanvasElement, assetMetadata: IAssetMetadata): Promise<IAssetMetadata> {
+        Guard.null(canvas);
+        Guard.null(assetMetadata);
+
+        // If the canvas or asset are invalid return asset metadata
+        if (!(canvas.width && canvas.height && assetMetadata.asset && assetMetadata.asset.size)) {
+            return assetMetadata;
+        }
+
+        await this.ensureModelLoaded();
+
+        const xRatio = assetMetadata.asset.size.width / canvas.width;
+        const yRatio = assetMetadata.asset.size.height / canvas.height;
+        const predictedRegions = await this.objectDetection.predictImage(
+            canvas,
+            this.settings.predictTag,
+            xRatio,
+            yRatio,
+        );
+
+        const updatedRegions = [...assetMetadata.regions];
+        predictedRegions.forEach((prediction) => {
+            const matchingRegion = updatedRegions.find((region) => {
+                return region.boundingBox
+                    && region.boundingBox.left === prediction.boundingBox.left
+                    && region.boundingBox.top === prediction.boundingBox.top
+                    && region.boundingBox.width === prediction.boundingBox.width
+                    && region.boundingBox.height === prediction.boundingBox.height;
+            });
+
+            if (updatedRegions.length === 0 || !matchingRegion) {
+                updatedRegions.push(prediction);
+            }
+        });
+
+        return {
+            ...assetMetadata,
+            regions: updatedRegions,
+            asset: {
+                ...assetMetadata.asset,
+                state: updatedRegions.length > 0 ? AssetState.Tagged : AssetState.Visited,
+                predicted: true,
+            },
+        } as IAssetMetadata;
+    }
+
+    public async ensureModelLoaded(): Promise<void> {
+        if (this.modelLoaded) {
+            return Promise.resolve();
+        }
+
+        await this.loadModel();
+        this.modelLoaded = true;
+    }
+
+    private async loadModel() {
+        let modelPath = "";
+        if (this.settings.modelPathType === ModelPathType.Coco) {
+            if (isElectron()) {
+                const appPath = this.getAppPath();
+
+                if (Env.get() !== "production") {
+                    modelPath = appPath + "/cocoSSDModel";
+                } else {
+                    modelPath = appPath + "/../../cocoSSDModel";
+                }
+            } else {
+                // TODO: Move this to a different location
+                modelPath = "https://jmangiadiag.blob.core.windows.net/vottcontainer";
```

Comment:
Remember we need to replace this with the VoTT web server we have for the web app.  The default Coco model should be deployed just there.

Reply:
Probably something like https://vott.z5.web.core.windows.net/cocoSSDModel

Reply:
Is that just an open file? What happens if that blob goes away? Any way to just host it directly in GitHub?

Reply:
@tbarlow12 it's a folder not a single file.  Blob storage is temporary.  Plan is to deploy this folder on the link above

Reply:
Moved to vott blob storage (same that host the website)